File: leetcode/125.valid-palindrome.py
# @leet start
import re


class Solution:
    def isPalindrome(self, s: str) -> bool:
        # Make lowercase and remove non-alphanumeric chars
        new_s = re.sub(r"[\W_]+", "", s.lower())

        # Check if the string is the same when it's reversed
        return new_s == new_s[::-1]


# @leet end

File: leetcode/125.valid-palindrome.ts
// @leet start
function isPalindrome(s: string): boolean {
  const alphanum = /[a-zA-Z\d]/;
  let i = 0;
  let j = s.length - 1;

  while (i < j) {
    if (!alphanum.test(s[i])) {
      i++;
      continue;
    }

    if (!alphanum.test(s[j])) {
      j--;
      continue;
    }

    if (s[i].toLowerCase() !== s[j].toLowerCase()) {
      return false;
    }

    i++;
    j--;
  }

  return true;
}
// @leet end
